### DISCUSSION TITLE
请问能否给deepin-terminal的提示符加上这种效果呢？

### DISCUSSION THREAD

**Discussion by xinguankeli** [linuxdeepin/developer-center#4696] [Show and tell | 成果展示]
https://github.com/linuxdeepin/developer-center/assets/96503070/f4728868-f4d5-4e0a-bd7c-c17a7a6d1df2

**Comment by BLumia** [comment]
没看懂，现在不就是这样吗？你输入和删除的时候会显示你当前输入光标位置

**Reply by xinguankeli** [reply]
这有个动画

**Reply by BLumia** [reply]
cc @ArchieMeng 

不过 QTermWidget 本身是没这个特性。以及我个人不太建议这种特性（在 TUI 应用里可能会看着很怪）

**Reply by xinguankeli** [reply]
这个会让terminal显得有质感

**Reply by xinguankeli** [reply]
这个要重做一个QlineEdit的功能，自己绘制这个提示符，我这里实现了但是有点卡

**Reply by BLumia** [reply]
> 以及我个人不太建议这种特性（在 TUI 应用里可能会看着很怪）

我的意思并不是说这种行为在实现上不可行，而是觉得这种行为可能更适合文本编辑器而不是终端模拟器。

**Reply by xinguankeli** [reply]
文本编辑器也行，不加终端加文本编辑器也行，反正都是提升质感

**Reply by BLumia** [reply]
> 反正都是提升质感

这个说法其实蛮主观的，并非所有人都会这么觉得... 我不清楚其他贡献者是否有兴趣去实现这个动效，我是觉得，如果某个编辑器提供了这种特性的话，我会倾向于把它关掉，所以我是没有意愿去实现它的 😂

**Reply by xinguankeli** [reply]
嗯 那就不加了

**Reply by ArchieMeng** [reply]
终端目前正在重构底层三方库，所以就算实现出来，也不太会立即接受这种PR.
文本编辑器的话可以试试，万一有人喜欢呢？不过，目前看情况，还是要优化你当前效果的性能。

**Reply by xinguankeli** [reply]
我后续就在文本编辑器上试一下吧，重构大概多久呢

**Reply by ArchieMeng** [reply]
说不好，如果想直接合入的话，就暂时推到ArchieMeng/deepin-terminal的update-termwidget分支。重构好了一起PR回去。

**Reply by xinguankeli** [reply]
不会是terminal了，可能会是编辑器

**Comment by xinguankeli** [comment]
有人关注吗

**Reply by Decodetalkers** [reply]
说实话，终端可能不适合这种效果

**Reply by xinguankeli** [reply]
那就加到文本编辑器呢？
